rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java

```diff
@@ -51,4 +51,32 @@ private static <R extends Mergeable<R>, E> R loadCvs(InputStream input, URI sour
         }
     }
 
+    private static String[] parseLine(String line) {
+        if (line.isEmpty()) {
+            return new String[0];
+        }
+
+        ArrayList<String> result = new ArrayList<>();
+        for (int i = 0; i < line.length(); i++) {
+            if (line.charAt(i) == ',') {
+                result.add("");
+                continue;
+            }
+            int valueStart = i;
+            char stopAt = ',';
+            if (line.charAt(i) == '"') {
+                valueStart = ++i;
+                stopAt = '"';
+            }
+            while (i < line.length() && line.charAt(i) != stopAt) {
+                i++;
+            }
+            result.add(line.substring(valueStart, i));
+            if (stopAt == '"') {
+                i++;
+            }
+        }
+        return result.toArray(new String[0]);
+    }
```

Comment:
Ah, okay, interesting. This works I suppose.

Reply:
The simple `String#split()` doesn't work anymore due to the quoted values and I didn't want to bring in a new Jackson dependency just for this. But like you noticed it isn't pretty and by no means optimized. Feel free to improve :-)